### ISSUE TITLE
no-useless-rename has false negative

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6502]
**What version of ESLint are you using?**

- v2.13.1

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

    $ echo "const {a, b: b} = obj" | eslint --stdin --env es6 --rule no-useless-rename:error --no-eslintrc

Or in our online demo:

![image](https://cloud.githubusercontent.com/assets/1937871/16256803/85b66bf8-388f-11e6-8c8c-110fd7bffb52.png)

**What did you expect to happen?**

```
<text>
  1:11  error  Destructuring assignment b unnecessarily renamed  no-useless-rename

✖ 1 problem (1 error, 0 warnings)
```

**What actually happened? Please include the actual, raw output from ESLint.**

No errors
